### PR TITLE
added yarn test to travis.yml

### LINKED ISSUE #5410: browser: use jest and enzyme to unit test react components
[Jest](https://facebook.github.io/jest/) is a javascript unit testing framework. It's often used for testing React components, but it's also a pretty good general purpose testing framework. [Enzyme](http://airbnb.io/enzyme/) is a JavaScript Testing utility for React that makes it easier to assert, manipulate, and traverse React Component's output.

## Expected Behavior
Write unit tests in jest to test javascript and react components

## Current Behavior
Unit tests only available for jsonrpc.js and it uses mocha.

## Possible Solution
1. Add unit test frameworks jest and enzyme
2. write unit tests for javascript and react components
3. run unit tests on CI and throw errors if there are any failures

## Steps to Reproduce (for bugs)
This is not a bug

## Context
Unit tests should cover all the javascript and react components.

Some of the advantages of Jest over Mocha are,
- ease of setup and configuration
- parallelizes test runs across workers to maximize performance
- fast interactive watch mode can run only test files related to changed files
- has a built-in code coverage reports which are a breeze to set up
- support for custom reporters

## Your Environment
master

### PR DESCRIPTION
**Pull request by kmkanagaraj**
## Description
Run `yarn test` in travis for every PR.

## Motivation and Context
Fixes #5410 

## How Has This Been Tested?

By running travis on https://github.com/kmkanagaraj/minio/tree/travis and seeing the results in https://travis-ci.org/kmkanagaraj/minio/builds/357244820
## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [ ] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have added unit tests to cover my changes.
- [ ] I have added/updated functional tests in [mint](https://github.com/minio/mint). (If yes, add `mint` PR # here: )
- [x] All new and existing tests passed.

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -14,13 +14,17 @@ os:
 env:
 - ARCH=x86_64
 
+before_install:
+- nvm install 8.10.0
```
Comment (harshavardhana) [inline .travis.yml]: Can we not use `nvm use stable`  any particular reason for `8.10.0` ?

Comment (kmkanagaraj) [reply .travis.yml]: There is no particular reason, will try with `nvm use stable`

**Comment by balamurugana** [comment]
## Mint Automation
| Test | Result |
|:--- |:--- |
| mint-xl.sh | :heavy_check_mark: |
| mint-fs.sh | :heavy_check_mark: |
| mint-dist-xl.sh | :heavy_check_mark: |
| mint-gateway-s3.sh | :heavy_check_mark: |
| mint-gateway-azure.sh | :heavy_check_mark: |

**Comment by balamurugana** [comment]
## Mint Automation
| Test | Result |
|:--- |:--- |
| mint-xl.sh | :heavy_check_mark: |
| mint-fs.sh | :heavy_check_mark: |
| mint-dist-xl.sh | :heavy_check_mark: |
| mint-gateway-s3.sh | :heavy_check_mark: |
| mint-gateway-azure.sh | :heavy_check_mark: |

**Comment by harshavardhana** [comment]
@kmkanagaraj looks like the tests have failed on the latest `stable` 

```
Summary of all failing tests
 FAIL  app/js/objects/__tests__/actions.test.js
  ● Objects actions › Download object › should download the object non-LoggedIn users
    TypeError: Cannot read property 'origin' of undefined
      268 |     } else {
      269 |       const url = `${
    > 270 |         window.location.origin
      271 |       }${minioBrowserPrefix}/download/${currentBucket}/${encObjectName}?token=''`
      272 |       window.location = url
      273 |     }
      
      at app/js/objects/actions.js:270:23
      at Object.dispatch (node_modules/redux-thunk/lib/index.js:11:18)
      at Object.<anonymous> (app/js/objects/__tests__/actions.test.js:451:13)
  ● Objects actions › Download object › should download the object for LoggedIn users
    TypeError: Cannot read property 'origin' of undefined
      473 |       return store.dispatch(actionsObjects.downloadObject("obj1")).then(() => {
      474 |         const url = `${
    > 475 |           window.location.origin
      476 |         }${minioBrowserPrefix}/download/bk1/${encodeURI(
      477 |           "pre1/obj1"
      478 |         )}?token=test`
      
      at app/js/objects/__tests__/actions.test.js:475:25
  ● Objects actions › should download checked objects
    TypeError: Cannot read property 'origin' of undefined
      565 |     return store.dispatch(actionsObjects.downloadCheckedObjects()).then(() => {
      566 |       const requestUrl = `${
    > 567 |         location.origin
      568 |       }${minioBrowserPrefix}/zip?token=test`
      569 |       expect(open).toHaveBeenCalledWith("POST", requestUrl, true)
      570 |       expect(send).toHaveBeenCalledWith(
      
      at app/js/objects/__tests__/actions.test.js:567:16
Test Suites: 1 failed, 49 passed, 50 total
Tests:       3 failed, 215 passed, 218 total
Snapshots:   0 total
Time:        12.694s
Ran all test suites.
error Command failed with exit code 1.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
The command "cd browser && yarn && yarn test && cd .." exited with 1.
```

**Comment by kmkanagaraj** [comment]
> @kmkanagaraj looks like the tests have failed on the latest stable

@harshavardhana yes, i am fixing those tests

**Comment by kmkanagaraj** [comment]
@harshavardhana fixed the failing tests, please review again.

**Pull request merged**